### PR TITLE
Finalize EnvironmentVariableMutatorOptions API

### LINKED ISSUE #179476: Allow EnvironmentVariableCollection API to apply changes via shell integration
A problem the Python extension has is that it sets environment variables but they get overridden during shell startup scripts. This issue would allow adding an option to the variable mutator for the timing a particular environment variable is applied.

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes #179476

### CODE REVIEW

File: src/vscode-dts/vscode.d.ts [hunk 1]
```diff
@@ -11376,8 +11397,10 @@ declare module 'vscode' {
 		 *
 		 * @param variable The variable to replace.
 		 * @param value The value to replace the variable with.
+		 * @param options Options applied to the mutator, when no options are provided this will
+		 * default to `{ applyAtProcessCreation: true }`.
```
Comment (karrtikr) [inline src/vscode-dts/vscode.d.ts:11401]: We should probably add a comment explaining default value of applyAtProcessCreation is actually false if any other parameter is specified, that does not seem obvious to me.

File: src/vs/workbench/api/common/extHostTerminalService.ts [hunk 2]
```diff
@@ -940,23 +940,14 @@ class UnifiedEnvironmentVariableCollection {
 	}
 
 	replace(variable: string, value: string, options: vscode.EnvironmentVariableMutatorOptions | undefined, scope: vscode.EnvironmentVariableScope | undefined): void {
-		if (this._extension && options) {
```
Comment (karrtikr) [inline src/vs/workbench/api/common/extHostTerminalService.ts:943]: `this._extension` property can be removed from constructor as well.

**Review by karrtikr — COMMENTED** [review]
CI failing

File: src/vscode-dts/vscode.d.ts (hunk 1 above)
Comment (Tyriar) [reply src/vscode-dts/vscode.d.ts:11402]: Opps, did it this morning but forgot to push

**Review by karrtikr — APPROVED** [review]

**Pull request merged**